portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -53,9 +55,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = request.getParameter("text-input");
+     UserService userService = UserServiceFactory.getUserService();
+     Entity commentEntity = new Entity("Comments");
+     String username;
+    if (userService.isUserLoggedIn()){
```

Comment:
If someone isn't logged in then you probably want to prevent them from leaving a comment here. So maybe return an error instead of comments.